### PR TITLE
Jop chardonnay

### PR DESCRIPTION
**Pull request by ldragoon**
1. Should no longer have issues w/blank nodes - #12 + #11 
2. json paths are configurable and have a configurable location

### CODE REVIEW

File: client/public/elements/app-moments-dropdown.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      if ( this.momentSelectEle.options[i].value === topic.moment ) {
+        this.momentSelectEle.selectedIndex = i;
+      };
+    }
+  }
+
+  async updated(changedProperties) {
+    if ( changedProperties.type === 'change' ) {
+      // Clear the store BEFORE reloading it or it loads them both in at the same time
+      this.MomentModel.store.data = new Object();
```
Comment (jrmerz) [inline client/public/elements/app-moments-dropdown.js]: @ldragoon we need to chat about this.  You should never directly edit the a stores data field.

File: client/public/elements/views/app-about.tpl.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+    main {
+      padding: 10px;
+    }
+  </style>
+
+  <main>
+    <h1>About IntertWINE</h1>
+    <p>
+      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt
```
Comment (jrmerz) [inline client/public/elements/views/app-about.tpl.js:21]: Do we have real copy for this?

File: client/public/elements/views/map/app-map-info-panel.js [hunk 3]
```diff
@@ -164,35 +252,68 @@ export default class AppMapInfoPanel extends Mixin(LitElement)
...
+      let check;
       for( let id in this.graph.links ) {
         link = this.graph.links[id];
-        if( link.src === node.id ) {
-          connections.push({
-            link,
-            node : this.graph.nodes[link.dst]
-          });
-        } else if ( link.dst === node.id ) {
+        if ( link['@id'].includes(node['@id']) ) {
```
Comment (jrmerz) [inline client/public/elements/views/map/app-map-info-panel.js]: @ldragoon why is the '@id' property an array here?

File: client/public/elements/views/map/app-map-info-panel.js [hunk 4]
```diff
@@ -164,35 +252,68 @@ export default class AppMapInfoPanel extends Mixin(LitElement)
...
+              node: this.graph.nodes[link.dst]
+            });
+          }
+        }
+      }
+
+      // TODO: Remove description-less connections that have links missing descriptions
+      // Those seem to mostly be verse terse dupes of main links
+      // There are still a few other dupes but this seems to take care of a lot of them
+      connections = connections.filter(connection => connection.link['description'] !== undefined);
```
Comment (jrmerz) [inline client/public/elements/views/map/app-map-info-panel.js]: Lets chat about this duplicate filtering

File: client/public/src/models/AppStateModel.js [hunk 5]
```diff
@@ -19,26 +19,37 @@ class AppStateModelImpl extends AppStateModel {
...
+    if ( state.location ) {
       let page = state.location.path ? state.location.path[0] : 'map';
       state.page = page || 'map';
+      state.first = true;
+    }
+
+    //state.moment = state.id || 'jop';
+
+    if ( state.first ) {
+      state.moment = 'jop';
```
Comment (jrmerz) [inline client/public/src/models/AppStateModel.js]: @ldragoon why hard coded 'jop'?  What is this state.first?

@jrmerz - the original has a hard-coded value that's the first moment we start with, I wasn't able to complete the "switch" process so a user can view both moments before Quinn had me push the code for you to review which will pull the json from Fedora.

File: config.js [hunk 6]
```diff
@@ -4,15 +4,17 @@ let clientPackage = require('./client/public/package.json');
 module.exports = {
   server : {
     assets : (env === 'prod') ? 'dist' : 'public',
-    appRoutes : ['map', 'moment', 'about'],
+    endpoint : 'https://sandbox.dams.library.ucdavis.edu/fcrepo/rest/collection/moments',
```
Comment (jrmerz) [inline config.js]: This should be an env variable.  You can make this the default if you want

File: controllers/api/graph.js [hunk 7]
```diff
@@ -1,10 +1,26 @@
...
-  let data = fs.readFileSync(path.join(__dirname, '..', '..', 'mock', 'mock-graph.json'), 'utf-8');
-  res.json(JSON.parse(data));
+const config   = require('../../config');
+const endpoint = config.server.endpoint;
+
+router.get('/:id', (req, res) => {
+  const url = endpoint + '/' + req.params.id + '/' + req.params.id + '.json';
+
+  // https://daveceddia.com/unexpected-token-in-json-at-position-0/
+  fetch(url)
```
Comment (jrmerz) [inline controllers/api/graph.js]: Switch to await statements

File: controllers/static.js [hunk 8]
```diff
@@ -13,20 +14,30 @@ const bundle = `
...
     app: app, // pass the express app
     htmlFile : path.join(assetsDir, 'index.html'), // pass the file you want to use
     isRoot : true, // are we serving from host root (/)?
-    appRoutes : config.server.appRoutes, // array of root paths.  ie appRoutes = ['foo', 'bar'] to server /foo/* /bar/*
+    appRoutes : config.server.appRoutes, // array of root paths. ie appRoutes = ['foo', 'bar'] to server /foo/* /bar/*
     getConfig : async (req, res, next) => {
       next({
         appRoutes : config.server.appRoutes,
+        mocks : mocks,
+        endpoint : 'https://sandbox.dams.library.ucdavis.edu/fcrepo/rest/collection/moments',
```
Comment (jrmerz) [inline controllers/static.js]: Why hardcoded again?  This should be pulled from config and should be a environment variable.

File: server.js [hunk 9]
```diff
@@ -1,10 +1,28 @@
...
+const moments  = config.server.moments;
+
+// TODO: Implement this functionality once the json version match both here and in Fedora
+// Load the files locally on app start so we have a fall back in case they can't get to the server
+for ( let i = 0; i < moments.length; i++ ) {
+  let url = endpoint + '/' + moments[i] + '/' + moments[i] + '.json';
+
+  let file = fs.createWriteStream('./mock/' + moments[i] + '.json');
+
+  https.get(url, (res) => {
```
Comment (jrmerz) [inline server.js]: Switch to node-fetch

File: client/public/src/models/MomentModel.js [hunk 10]
```diff
@@ -26,39 +25,147 @@ class MomentModel extends BaseModel {
...
   transformMockLinks(data) {
-    let nodes = {};
-    let links = {};
-    let item;
-
-    for( item of data ) {
-      if( item.type === 'connection' ) {
-        item.id = item.src+'-'+item.dst;
-        links[item.id] = item;
+    let links = {}, nodes = {}, lookup = {};
```
Comment (jrmerz) [inline client/public/src/models/MomentModel.js:34]: @ldragoon let's review this together

File: client/public/elements/views/app-about.tpl.js (hunk 2 above)
Comment (ldragoon) [reply client/public/elements/views/app-about.tpl.js:21]: No copy yet for this. :(

File: client/public/elements/app-moments-dropdown.js (hunk 1 above)
Comment (ldragoon) [reply client/public/elements/app-moments-dropdown.js]: What's the best way to clear it out before swapping in the new json?

File: client/public/elements/views/map/app-map-info-panel.js (hunk 3 above)
Comment (ldragoon) [reply client/public/elements/views/map/app-map-info-panel.js]: I think I am able to remove this code now. Can we review the update?

File: client/public/elements/views/map/app-map-info-panel.js (hunk 4 above)
Comment (ldragoon) [reply client/public/elements/views/map/app-map-info-panel.js]: Commented this out, seems like it may no longer be necessary.

**Pull request merged**